README.md
# quant-methods-elites
**Quantitative Methods for Studying Elites: Demonstration for R**

[![Version](https://img.shields.io/badge/version-v1.0.0-blue.svg)](CHANGELOG.md) [![Project Status: Inactive – The project has reached a stable, usable state but is no longer being actively developed; support/maintenance will be provided as time allows.](https://www.repostatus.org/badges/latest/inactive.svg)](STATUS.md) [![GitHub issues](https://img.shields.io/github/issues/bgonzalezbustamante/quant-methods-elites.svg)](https://github.com/bgonzalezbustamante/quant-methods-elites/issues/) [![GitHub issues-closed](https://img.shields.io/github/issues-closed/bgonzalezbustamante/quant-methods-elites.svg)](https://github.com/bgonzalezbustamante/quant-methods-elites/issues?q=is%3Aissue+is%3Aclosed) [![DOI](https://zenodo.org/badge/455969529.svg)](https://zenodo.org/badge/latestdoi/455969529) [![License](https://img.shields.io/badge/license-GNU%20GPLv3-black)](LICENSE-GPL.md) [![License](https://img.shields.io/badge/license-CC%20BY%204.0-black)](LICENSE-CC.md) [![R](https://img.shields.io/badge/made%20with-R%20v4.1.0-1f425f.svg)](https://cran.r-project.org/) [![Latex](https://img.shields.io/badge/made%20with-LaTeX-1f425f.svg)](https://www.latex-project.org/)

## Overview

This repository contains demonstrations for `R` of generalised linear models and proportional hazards Cox regressions to measure the impact of individual business trajectories on the access and permanence in political positions in the Chilean executive and legislative branches. These demonstrations are the primary analysis of the forthcoming methodological article entitled “*Métodos cuantitativos para estudiar a las élites: Aplicaciones prácticas, sesgos y potencialidades*”.[^1]

In addition, it contains an anonymised, sliced data set from the Chilean Elite Survey (1990-2010) (*N* = 386) in Comma-Separated Values `CSV` format with Unicode encoding `UTF-8` based on the latest update by Joignant and González-Bustamante (2014).[^2] The second data set used in the demonstrations is González-Bustamante and Olivares (2022).[^3]

## Metadata and Preservation

This code and data are stored with version control on a GitHub repository. Furthermore, a Digital Object Identifier is provided by Zenodo.

``` r
quant-methods-elites
|-- .Rprofile
|-- .gitignore
|-- CHANGELOG.md
|-- CITATION.cff
|-- CODE_OF_CONDUCT.md
|-- LICENSE-CC.md
|-- LICENSE-GPL.md
|-- README.md
|-- STATUS.md
|-- quant-methods-elites.Rproj
|-- renv.lock
|-- code
    |-- stage_0_descriptives.R
    |-- stage_1_data_cleaning.R
    |-- stage_2_glm.R
    |-- stage_3_cox.R
|-- data
    |-- raw
        |-- chamber_1990_1994.csv
        |-- chamber_1994_1998.csv
        |-- chamber_1998_2002.csv
        |-- chamber_2002_2006.csv
        |-- chamber_2006_2010.csv
        |-- chamber_2010_2014.csv
    |-- tidy
        |-- elite_survey_2010.csv
|-- refs
    |-- BIB-QM-Elites.bib
|-- renv
    |-- .gitignore
    |-- activate.R
    |-- settings.dcf
|-- results
    |-- figures
        |-- figure_01a.png
        |-- figure_01b.png
        |-- figure_02.png
        |-- figure_03a.jpg
        |-- figure_03b.jpg
        |-- figure_03c.jpg
        |-- figure_04.jpg
    |-- tables
        |-- table_01.html
        |-- table_01.tex
        |-- table_02.html
        |-- table_02.tex
        |-- table_03.html
        |-- table_03.tex
```

9 directories and 39 files.

In addition, this README file in Markdown `MD` format provides specific information to ensure the replicability of the code.

## Storage and Backup

The GitHub repository has controlled access with Two-Factor Authentication `2FA` with two physical USB security devices (Bastián González-Bustamante, [ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820)). Both USB keys issue one-time passwords to generate a cryptographic authentication `FIDO2` and `U2F`.

Moreover, the repository is backed up on Hierarchical File Server `HFS` for recovery in case of incidents. This backup is located on the University of Oxford hub connected to `Code42` Cloud Backup encrypted with `256-bit AES`. The backup is performed with every change on GitHub and receives weekly light maintenance and a deep one every month. This backup will be secured until May 2024. An extension of this period will be evaluated on budget availability.

## Getting Started

### Software

We used `R v4.1.0 -- Camp Pontanezen` and the following packages: `foreign v0.8-81`, `performance v0.8.0`, `pscl v1.5.5`, `sjPlot v2.8.9`, `stargazer v5.2.2`,	`survival v3.2-11` and `tidyverse v1.3.1`. In addition, we have used `ggplot2 v3.3.3` as a dependency.

Propensity score estimation, under embargo, was carried out using `MatchIt v4.2.0`, `cobalt v4.3.1`, `lmtest v0.9-38`, `sandwich v3.0-1` and, as a dependency, `optmatch v0.9-14`. A more complex code is already released as supplementary material in González-Bustamante (2022).[^4]

We recommend that users run replication code and scripts from the root directory using the `R` project `quant-methods-elites.Rproj` or, if they wish, import the anonymised, sliced data set from the Chilean Elite Survey (1990-2010) directly using the code below. The instructions for importing data on cabinets are available on this [**GitHub repository**](https://github.com/bgonzalezbustamante/chilean-ministers/).

### Import Data

#### R Code

``` r
## GitHub Repository
github_1 <- "https://raw.githubusercontent.com/"
github_2 <- "bgonzalezbustamante/quant-methods-elites/main/data/tidy/"

## Elite Survey Data
elite_survey <- read.csv(paste(github_1, github_2, "elite_survey_2010.csv", sep = ""),
                              header = T, sep = ",", encoding = "UTF-8")
```
#### Python Code

``` python
import pandas as pd

## GitHub Repository
github_1 = "https://raw.githubusercontent.com/"
github_2 = "bgonzalezbustamante/quant-methods-elites/main/data/tidy/"

## Elite Survey Data
url = github_1 + github_2 + "elite_survey_2010.csv"
df = pd.read_csv(url, index_col=0)
```

### Replication Instructions

Folder `code` contains the `R` scripts.

Subfolders `results/figures` and `results/tables` include all plots and tables provided as `PNG`, `JPG`, `HTML`, and `TeX` files, respectively.

- **Stage 1.** Please do not run `R` script `stage_1_data_cleaning.R` from the "code" folder since it requires complete, non-anonymised data sets under embargo.

- **Stage 2.** Run script `stage_2_glm.R` from the `code` folder. This script contains GLMs demonstrations for `R`. Model after matching is not available for the moment since the propensity score chunk is still under embargo.

- **Stage 3.** Run script `stage_3_cox.R` from the `code` folder. This script contains proportional hazards models for `R`. Model after matching is not available for the moment since the propensity score chunk is still under embargo.

The file `stage_0_descriptives.R` contains a descriptive code for the Chamber of Deputies using a sliced data set from González-Bustamante and Cisternas (2016).[^5] This is not necessary for the models.

To ensure the replicability of our analyses, we used project-local R dependency with `renv v0.16.0`.

``` r
library(renv)
renv::init()
renv::install("tidyverse@1.3.1")
renv::install("foreign@0.8-81")
renv::install("performance@0.8.0")
renv::install("pscl@1.5.5")
renv::install("stargazer@5.2.2")
renv::install("sjPlot@2.8.9")
renv::install("ggplot2@3.3.3")
renv::install("MatchIt@4.2.0") ## PSA code under embargo
renv::install("cobalt@4.3.1") ## PSA code under embargo
renv::install("lmtest@0.9-38") ## PSA code under embargo
renv::install("sandwich@3.0-1") ## PSA code under embargo
renv::install("optmatch@0.9-14") ## PSA code under embargo
renv::install("survival@3.2-11")
renv::snapshot()
renv::status()
```

### Codebook

The file `elite_survey_2010.csv` in `data/tidy` subfolder is the anonymised, sliced data set on Chilean elite between 1990 and 2010. This set contains 386 observations.

- `id`. Unique ID per case.

- `sex`. Case sex.

- `dob`. Date of birth.

- `political_capital`. A nominal variable indicating the highest post in a political party: national, regional, local level, none formal position.

- `education`. A nominal variable indicating the highest educational level: primary school, high school, BA/BPS completed, technical career achieved, unfinished college or technical career, graduate studies without a degree, MA/MBA, PhD, PhD candidate.

- `school`. A nominal variable indicating the type of primary school: secular private, religious private, subsidised private, local government school, or public primary school.

- `family_capital`. A dummy variable indicating whether the case's parents held government first-line positions (*i.e.*, minister, undersecretary or senior public manager) or was a congressman or congresswoman.

- `capital_father`. A dummy variable indicating whether the case's father held government first-line positions (*i.e.*, minister, undersecretary or senior public manager) or was a congressman or congresswoman.

- `capital_mother`. A dummy variable indicating whether the case's mother held government first-line positions (*i.e.*, minister, undersecretary or senior public manager) or was a congressman or congresswoman.

- `party_1`. A nominal variable indicating partisanship in chronological order (first option).

- `party_2`. A nominal variable indicating partisanship in chronological order (second option).

- `party_3`. A nominal variable indicating partisanship in chronological order (third option).

- `party_4`. A nominal variable indicating partisanship in chronological order (fourth option).

- `t1`. A nominal variable indicating public and private careers in chronological order (first post).

- `t2`. A nominal variable indicating public and private careers in chronological order (second post).

- `t3`. A nominal variable indicating public and private careers in chronological order (third post).

- `t4`. A nominal variable indicating public and private careers in chronological order (fourth post).

- `t5`. A nominal variable indicating public and private careers in chronological order (fifth post).

- `t6`. A nominal variable indicating public and private careers in chronological order (sixth post).

- `t7`. A nominal variable indicating public and private careers in chronological order (seventh post).

- `t8`. A nominal variable indicating public and private careers in chronological order (eighth post).

- `t9`. A nominal variable indicating public and private careers in chronological order (ninth post).

- `exp_business`. A dummy variable indicating whether the case has a previous trajectory as a CEO.

- `president`. A dummy variable indicating whether the case's highest political post was president.

- `minister`. A dummy variable indicating whether the case's highest political post was minister.

- `senator`. A dummy variable indicating whether the case's highest political post was senator.

- `deputy`. A dummy variable indicating whether the case's highest political post was deputy.

- `undersecretary`. A dummy variable indicating whether the case's highest political post was undersecretary (*i.e.*, viceminister).

- `intendant`. A dummy variable indicating whether the case's highest political post was intendant.

- `ceo`. A dummy variable indicating whether the case's highest political post was senior public manager.

- `cabinet_chief`. A dummy variable indicating whether the case's highest political post was cabinet chief.

- `party_leader`. A dummy variable indicating whether the case's highest political post was party leader.

## License

The content of this project itself is licensed under a [Creative Commons Attribution 4.0 International license (CC BY 4.0)](LICENSE-CC.md), and the underlying code used to format and display that content is licensed under an [GNU GPLv3 license](LICENSE-GPL.md).

The above implies that the data set may be shared, reused, and adapted as long as appropriate acknowledgement is given. In addition, the code may be shared, reused, and adapted as long as the source is disclosed, the changes are stated, and the same [GNU GPLv3 license](LICENSE-GPL.md) is used.

## Contribute

Contributions are entirely welcome. You just need to [open an issue](https://github.com/bgonzalezbustamante/quant-methods-elites/issues/new) with your comment or idea.

For more substantial contributions, please fork this repository and make changes. Pull requests are also welcome.

Please read our [code of conduct](CODE_OF_CONDUCT.md) first. Minor contributions will be acknowledged, and significant ones will be considered on our contributor roles taxonomy.

## Citation

González-Bustamante, B. (2022). Quantitative Methods for Studying Elites: Demonstration for R (Version 1.0.0 -- Quiet Hill) [Computer software]. DOI: [10.5281/zenodo.6098061](https://doi.org/10.5281/zenodo.6098061)

## Author

Bastián González-Bustamante \
bastian.gonzalezbustamante@politics.ox.ac.uk \
[ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820) \
https://bgonzalezbustamante.com

## CRediT - Contributor Roles Taxonomy

Bastián González-Bustamante ([ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820)): Conceptualisation, data curation, formal analysis, methodology, resources, software, validation, visualisation, writing – original draft, and writing – review and editing.

Alfredo Joignant ([ORCID iD 0000-0002-5811-0988](https://orcid.org/0000-0002-5811-0988)): Data curation, funding acquisition, project administration, resources, supervision, and validation.

Gonzalo Delamaza ([ORCID iD 0000-0002-5418-3135](https://orcid.org/0000-0002-5418-3135)): Resources and supervision.

Hernán Cuevas ([ORCID iD 0000-0002-4295-5652](https://orcid.org/0000-0002-4295-5652)): Resources and supervision.

Manuel Gárate ([ORCID iD 0000-0002-0016-596X](https://orcid.org/0000-0002-0016-596X)): Resources and supervision.

ANID/FONDECYT 1100877 Research Team: Investigation.[^6]

### Latest Revision

[November 25, 2022](CHANGELOG.md).

[^1]: González-Bustamante, B. (2022, *forthcoming*). Métodos cuantitativos para estudiar a las élites: Aplicaciones prácticas, sesgos y potencialidades. *Revista Chilena de Derecho y Ciencia Política*. SocArXiv: [10.31235/osf.io/5m2ur](https://doi.org/10.31235/osf.io/5m2ur).
[^2]: Joignant, A., & González-Bustamante, B. (2014). El núcleo de la élite política chilena 1990-2010. Proyecto Fondecyt 1100877. Working Paper, Universidad Diego Portales.
[^3]: González-Bustamante, B., & Olivares, A. (2022). Data Set on Chilean Ministers (1990-2014) (Version 3.3.6 -- Dry Bonus) [Data set]. DOI: [10.5281/zenodo.5744536](https://doi.org/10.5281/zenodo.5744536).
[^4]: González-Bustamante, B. (2022). Ministerial Stability During Presidential Approval Crises: The Moderating Effect of Ministers’ Attributes on Dismissals in Brazil and Chile. *The British Journal of Politics and International Relations*. OnlineFirst. DOI: [10.1177/13691481221124850](https://doi.org/10.1177/13691481221124850).
[^5]: González-Bustamante, B., & Cisternas (2016). Élites políticas en el poder legislativo chileno: La Cámara de Diputados (1990-2014). *Política, Revista de Ciencia Política, 54*(1), 19-52.
[^6]: For contributions related to cabinet data, see this [GitHub repository](https://github.com/bgonzalezbustamante/chilean-ministers/).


CHANGELOG.md
# Changelog
All notable changes to "Quantitative Methods for Studying Elites: Demonstration for R" will be documented in this file.

## [Unreleased]

## [2.0.0] - TBC
### Added
- Code under embargo.

## [Released]

## [1.0.0] - 2022-11-25
### Added
- Project-local R dependency.
### Changed
- Minor corrections.

## [0.13.11] - 2022-11-16
### Changed
- BibTeX file.
- Minor corrections.

## [0.13.10] - 2022-07-19
### Changed
- Minor corrections.

## [0.13.9] - 2022-06-23
### Changed
- Minor corrections.

## [0.13.8] - 2022-06-18
### Changed
- Minor corrections.

## [0.13.7] - 2022-04-05
### Changed
- Minor corrections.

## [0.13.6] - 2022-04-01
### Changed
- Minor corrections.

## [0.13.5] - 2022-03-31
### Added
- GNU GPLv3 license on GitHub repository.
### Changed
- Minor corrections.
### Removed
- MIT license on GitHub repository.

## [0.12.5] - 2022-03-17
### Added
- MIT license on GitHub repository.
### Changed
- Minor corrections.
### Deprecated
- MIT license on GitHub repository.

## [0.11.5] - 2022-02-20
### Added
- Quantitative methods for studying elites preprint on private OSF component.
### Changed
- BibTeX file.

## [0.10.5] - 2022-02-19
### Changed
- BibTeX file.

## [0.10.4] - 2022-02-18
### Added
- Sliced data set of Chamber of Deputies from González-Bustamante and Cisternas (2016) in CSV UTF-8 format.
- Descriptives of Chamber cases code.
### Changed
- GLMs code.
- Cox models code.
- BibTeX file.
- Minor corrections.

## [0.9.4] - 2022-02-16
### Changed
- Cox models code.
- Minor corrections.

## [0.9.3] - 2022-02-15
### Added
- Zenodo DOI.
- Citation CFF file.
### Changed
- Revision of upgrade versioning.
- BibTeX file.
- Minor corrections.

## [0.8.3] - 2022-02-12
### Added
- Import data code.
### Changed
- Revision of upgrade versioning.
- BibTeX file.
- Minor corrections.

## [0.7.3] - 2022-02-10
### Added
- BibTeX file.
### Changed
- Minor corrections.

## [0.6.3] - 2022-02-07
### Added
- Cox models code.
- Propensity score analysis code (under embargo).
### Changed
- GLMs code.

## [0.5.3] - 2022-02-06
### Added
- GLMs code.
### Changed
- Elite survey data cleaning code.
- Anonymised, sliced data set in CSV UTF-8 format.

## [0.4.3] - 2022-02-05
### Added
- GitHub repository.
- README in MD format on GitHub repository.
- Elite survey data cleaning code.
- Anonymised, sliced data set in CSV UTF-8 format.
- Changelog in MD format on GitHub repository.
- License CC BY 4.0 on GitHub repository.
- Badges in SVG on GitHub repository.
- Code of conduct in MD format.
### Removed
- Complete, non-anonymised, revised data set in SAV format.

## [0.3.3] - 2021-07-20
### Added
- Extended abstract on private OSF component.
### Deprecated
- Complete, non-anonymised, revised data set in SAV format.

## [0.2.3] - 2014-09-24
### Added
- Complete, non-anonymised, revised data set in SAV format.
### Removed
- Complete, non-anonymised data set in SAV format.

## [0.1.3] - 2014-09-23
### Changed
- Complete, non-anonymised data set in SAV format.
### Deprecated
- Complete, non-anonymised data set in SAV format.

## [0.1.2] - 2014-09-20
### Changed
- Complete, non-anonymised data set in SAV format.

## [0.1.1] - 2014-09-19
### Changed
- Complete, non-anonymised data set in SAV format.

## [0.1.0] - 2014-09-13
### Added
- Complete, non-anonymised data set in SAV format.


code/stage_3_cox.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Proportional Hazards Models Code QMM Studying Elites
## R version 4.1.0 (2021-05-18) -- "Camp Pontanezen"
## Date: February 2022

## Bastián González-Bustamante (University of Oxford, UK)
## https://bgonzalezbustamante.com

## GitHub Repository
## github.com/bgonzalezbustamante/quant-mixed-methods-elites
## DOI: 10.5281/zenodo.6098061

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Packages and Data ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Clean Environment
rm(list = ls())

## Packages
library(survival)
library(performance)
library(stargazer)
library(sjPlot)
library(ggplot2)

## GitHub Repository
github_1 <- "https://raw.githubusercontent.com/"
github_2 <- "bgonzalezbustamante/chilean-ministers/main/data/tidy/"

## Chilean Ministers Data
chilean_ministers <- read.csv(paste(github_1, github_2, "Chilean_cabinets_1990_2014.csv", sep = ""),
                              header = T, sep = ",", encoding = "UTF-8")

chilean_ministers$prestige <- ifelse(chilean_ministers$economist == 1, 1,
                                     ifelse(chilean_ministers$lawyer == 1, 1, 0))

sum(table(chilean_ministers$exp_business))
sum(table(chilean_ministers$sex))
sum(table(chilean_ministers$party_leader))
sum(table(chilean_ministers$prestige))
sum(table(chilean_ministers$political_kinship))

## Time Variables
chilean_ministers$time_minister <- with(chilean_ministers, (as.Date(chilean_ministers$end_minister)
                                                   - as.Date(chilean_ministers$start_minister)))

chilean_ministers$time_minister <- as.numeric(chilean_ministers$time_minister)

chilean_ministers$time_president <- with(chilean_ministers, (as.Date(chilean_ministers$end_president)
                                                   - as.Date(chilean_ministers$start_president)))

chilean_ministers$time_president <- as.numeric(chilean_ministers$time_president)

chilean_ministers$prop_term <- chilean_ministers$time_minister / chilean_ministers$time_president

## Event Variables
chilean_ministers$exit <- ifelse(chilean_ministers$end_minister < chilean_ministers$end_president, 1, 0)

chilean_ministers$t75 <- ifelse(chilean_ministers$prop_term >= 0.75, 1, 0)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Cox Proportional Hazards Models ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

cox_1 <- coxph(Surv(time_minister, t75 == 1) ~ exp_business + I(sex) + non_party + party_leader
               + prestige + political_kinship, ties = "breslow", data = chilean_ministers)

cox_2 <- coxph(Surv(time_minister, exit == 1) ~ exp_business + I(sex) + non_party + party_leader
               + prestige + political_kinship + I(president), ties = "efron", data = chilean_ministers)

## Code PSA (under embargo)
source("../secured-data/qm-elites/stage_3_psa_embargo.R", encoding = "UTF-8")

## Standardised and Absolute Mean Differences
## png("results/figures/figure_02.png", width = (1024*2), height = (768*2), units = 'px', res = 300)
love.plot(c_out_1, stat = "mean.diffs", poly = 1, abs = TRUE,
          drop.distance = TRUE, thresholds = c(m = .1),
          var.order = "unadjusted",
          shapes = c("square filled", "triangle filled"),
          colors = c("black", "black"),
          var.names = loveplot_var_3,
          sample.names = c("Sin Matching", "Full Matching"),
          line = FALSE, stars = "none", title = NULL) + theme_minimal(base_size = 12) +
  theme(plot.margin = unit(c(0.5,0.5,0.5,0.5), "cm"), legend.title = element_blank()) + xlab("Gabinete")
## dev.off()

## Check Multicollinearity
performance::check_collinearity(cox_1)
performance::check_collinearity(cox_2)

## PHA Test
res.zph1 <- cox.zph(cox_1)
res.zph2 <- cox.zph(cox_2)
res.zph3 <- cox.zph(cox_3)

stargazer(cox_1, cox_2, cox_3,
          type = "latex", header = FALSE, style = "ajps", out = "results/tables/table_03.tex",
          title = "Modelos de riesgos proporcionales de Cox y resultados después del matching para salidas del gabinete en Chile (1990-2014)",
          se = list(NULL, NULL, round(summary(cox_3)[7]$coefficients[4], 3)),
          dep.var.labels = c("Umbral 75%", "Salida censurada"), notes.align = "c",
          model.numbers = FALSE, omit = "president",
          column.labels = c("Modelo VIII","Modelo IX","Modelo X"),
          omit.stat = c("rsq", "max.rsq", "wald", "lr", "logrank"),
          add.lines = list(c("Matching", "No", "No", "Full"),
                           c("Sub. Clustering", "No", "No", "Sí"),
                           c("Efectos fijos (gobierno)", "No", "Sí", "PSA"),
                           c("Submuestra", "No", "No", "No"),
                           c("Concurrencia", "Breslow", "Efron", "Efron"),
                           c("PHA", round(res.zph1$table[7,3], 3), round(res.zph2$table[8,3], 3),
                             round(res.zph3$table[2,3], 3)),
                           c("VIF", "Bajo", "Bajo", "Bajo"),
                           c("AIC", round(extractAIC(cox_1)[2], 3),
                             round(extractAIC(cox_2)[2], 3), round(extractAIC(cox_2)[2], 3)),
                           c("C-Index", round(cox_1$concordance[6], 3),
                             round(cox_2$concordance[6], 3), round(cox_3$concordance[6], 3)),
                           c("Eventos", cox_1$nevent, cox_2$nevent, cox_3$nevent)),
          covariate.labels = c("Trayectoria empresarial", "Sexo (mujer)", "Independencia política",
                               "Dirigente de partido", "Profesión de prestigio", "Capital político familiar"))

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### HR Plot ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

plot_model(cox_3_plot, type = "pred", terms = c("exp_business")) +
  theme_classic() + coord_cartesian(expand = TRUE, ylim = c(0, 1)) +
  labs(x = "\nTrayectoria empresarial \nDespués de Full Matching", y = "Coeficiente de riesgo", caption = "") +
  ggtitle("") +
  theme(axis.title.x = element_text(face = "bold"),
        axis.title.y = element_text(face = "bold"),
        legend.title = element_blank(),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.7),
        panel.background = element_blank(),
        axis.line = element_blank()) +
  scale_x_continuous(breaks = c(0, 1),
                     labels = c("No posee", "Posee"))
ggsave("results/figures/figure_04.jpg", width = 4, height = 4, units = "in")


code/stage_1_data_cleaning.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Data Cleaning Code Chilean Elite Survey 1990-2010
## R version 4.1.0 (2021-05-18) -- "Camp Pontanezen"
## Date: February 2022

## Bastián González-Bustamante (University of Oxford, UK)
## https://bgonzalezbustamante.com

## GitHub Repository
## github.com/bgonzalezbustamante/quant-mixed-methods-elites
## DOI: 10.5281/zenodo.6098061

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Packages and Data ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Clean Environment
rm(list = ls())

## Packages
library(foreign)
library(tidyverse)

## Data
elite_data <- read.spss("../secured-data/Elites-Survey-2010/BASE ÉLITES EN CHILE_FONDECYT.sav", to.data.frame = TRUE)
revision <- read.csv("../secured-data/Elites-Survey-2010/20140924-Joignant-Gonzalez-Bustamante.csv")

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### González-Bustamante and Garrido-Vergara Variables ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

table(elite_data$CARGOS[which(elite_data$J_GB == 1)])
table(elite_data$P24_REC[which(elite_data$J_GB == 1)])

test1 <- c("Cargos de Primera Línea", "Parlamentarios", "Jefe de Servicio", "Directorio o Gerencia de Empresa Pública")
test2 <- "Cargos de Gobierno de Primera Línea"

elite_data$father <- ifelse(elite_data$P33.1_REC1 == test1[1] | elite_data$P33.1_REC1 == test1[2] | elite_data$P33.1_REC1 == test1[3] | elite_data$P33.1_REC1 == test1[4], 1,
                            ifelse(elite_data$P33.1_REC2 == test1[1] | elite_data$P33.1_REC2 == test1[2] | elite_data$P33.1_REC2 == test1[3] | elite_data$P33.1_REC2 == test1[4], 1,
                                   ifelse(elite_data$P33.1_REC3 == test1[1] | elite_data$P33.1_REC3 == test1[2] | elite_data$P33.1_REC3 == test1[3] | elite_data$P33.1_REC3 == test1[4], 1,
                                          ifelse(elite_data$P33.1_REC4 == test1[1] | elite_data$P33.1_REC4 == test1[2] | elite_data$P33.1_REC4 == test1[3] | elite_data$P33.1_REC4 == test1[4], 1,
                                                 ifelse(elite_data$P33.1_REC5 == test1[1] | elite_data$P33.1_REC5 == test1[2] | elite_data$P33.1_REC5 == test1[3] | elite_data$P33.1_REC5 == test1[4], 1,
                                                        ifelse(elite_data$P33.1_REC6 == test1[1] | elite_data$P33.1_REC6 == test1[2] | elite_data$P33.1_REC6 == test1[3] | elite_data$P33.1_REC6 == test1[4], 1,
                                                               ifelse(elite_data$P33.1_REC7 == test1[1] | elite_data$P33.1_REC7 == test1[2] | elite_data$P33.1_REC7 == test1[3] | elite_data$P33.1_REC7 == test1[4], 1,
                                                                      ifelse(elite_data$P33.1_REC8 == test1[1] | elite_data$P33.1_REC8 == test1[2] | elite_data$P33.1_REC8 == test1[3] | elite_data$P33.1_REC8 == test1[4], 1, 0))))))))

elite_data$mother <- ifelse(elite_data$P33.2_REC1 == test2 | elite_data$P33.2_REC1 == test1[2] | elite_data$P33.2_REC1 == test1[3] | elite_data$P33.2_REC1 == test1[4], 1,
                            ifelse(elite_data$P33.2_REC2 == test2 | elite_data$P33.2_REC2 == test1[2] | elite_data$P33.2_REC2 == test1[3] | elite_data$P33.2_REC2 == test1[4], 1,
                                   ifelse(elite_data$P33.2_REC3 == test2 | elite_data$P33.2_REC3 == test1[2] | elite_data$P33.2_REC3 == test1[3] | elite_data$P33.2_REC3 == test1[4], 1,
                                          ifelse(elite_data$P33.2_REC4 == test2 | elite_data$P33.2_REC4 == test1[2] | elite_data$P33.2_REC4 == test1[3] | elite_data$P33.2_REC4 == test1[4], 1,
                                                 ifelse(elite_data$P33.2_REC5 == test2 | elite_data$P33.2_REC5 == test1[2] | elite_data$P33.2_REC5 == test1[3] | elite_data$P33.2_REC5 == test1[4], 1,
                                                        ifelse(elite_data$P33.2_REC6 == test2 | elite_data$P33.2_REC6 == test1[2] | elite_data$P33.2_REC6 == test1[3] | elite_data$P33.2_REC6 == test1[4], 1,
                                                               ifelse(elite_data$P33.2_REC7 == test2 | elite_data$P33.2_REC7 == test1[2] | elite_data$P33.2_REC7 == test1[3] | elite_data$P33.2_REC7 == test1[4], 1, 0)))))))

elite_data$father[is.na(elite_data$father)] <- 0
elite_data$mother[is.na(elite_data$mother)] <- 0
elite_data$parents <- ifelse(elite_data$father == 1, 1, ifelse(elite_data$mother == 1, 1, 0))
table(elite_data$parents[which(elite_data$J_GB == 1)])

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Joignant and González-Bustamante Corrections ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

elite_core <- select(revision, PRESIDENTE, MINISTRO, SENADOR, DIPUTADO, SUBSECRETARIO,
                     INTENDENTE, AUTORID..ECO..REG., JEFES.GAB..O.DIV., MIEMBROS.MESAS.DIR..PP)
elite_core <- cbind(id = rownames(elite_core), elite_core)
elite_data <- cbind(id = rownames(elite_data), elite_data)
elite_data <- left_join(elite_data, elite_core, by = "id")

## Check
sum(elite_data$J_GB - elite_data$MINISTRO)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### CSV File ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

anonymised_data <- select(elite_data, id, Sexo, Año_nacimiento, CARGOS, P24_REC, P25.3_REC,
                          parents, father, mother, p5_1, p5_2, p5_3, p5_4, P6.1, P6.2, P6.3,
                          P6.4, P6.5, P6.6, P6.7, P6.8, P6.9, P6_REC2, PRESIDENTE, MINISTRO,
                          SENADOR, DIPUTADO, SUBSECRETARIO, INTENDENTE, AUTORID..ECO..REG.,
                          JEFES.GAB..O.DIV., MIEMBROS.MESAS.DIR..PP)
names(anonymised_data)[2] = "sex"
names(anonymised_data)[3] = "dob"
names(anonymised_data)[4] = "political_capital"
names(anonymised_data)[5] = "education"
names(anonymised_data)[6] = "school"
names(anonymised_data)[7] = "family_capital"
names(anonymised_data)[8] = "capital_father"
names(anonymised_data)[9] = "capital_mother"
names(anonymised_data)[10] = "party_1"
names(anonymised_data)[11] = "party_2"
names(anonymised_data)[12] = "party_3"
names(anonymised_data)[13] = "party_4"
names(anonymised_data)[14] = "t1"
names(anonymised_data)[15] = "t2"
names(anonymised_data)[16] = "t3"
names(anonymised_data)[17] = "t4"
names(anonymised_data)[18] = "t5"
names(anonymised_data)[19] = "t6"
names(anonymised_data)[20] = "t7"
names(anonymised_data)[21] = "t8"
names(anonymised_data)[22] = "t9"
names(anonymised_data)[23] = "exp_business"
names(anonymised_data)[24] = "president"
names(anonymised_data)[25] = "minister"
names(anonymised_data)[26] = "senator"
names(anonymised_data)[27] = "deputy"
names(anonymised_data)[28] = "undersecretary"
names(anonymised_data)[29] = "intendant"
names(anonymised_data)[30] = "ceo"
names(anonymised_data)[31] = "cabinet_chief"
names(anonymised_data)[32] = "party_leader"

## Save File
write.csv(anonymised_data, "data/tidy/elite_survey_2010.csv", fileEncoding = "UTF-8", row.names =  FALSE)
## write.csv(anonymised_data, "../women-trajectories/data/tidy/elite_survey_2010.csv",
          ## fileEncoding = "UTF-8", row.names =  FALSE)
